pkg/adsc/adsc.go

```diff
@@ -202,8 +203,35 @@ type ResponseHandler interface {
 	HandleResponse(con *ADSC, response *discovery.DiscoveryResponse)
 }
 
+// jsonMarshalProtoWithName wraps a proto.Message with name so it can be marshaled with the standard encoding/json library
+type jsonMarshalProtoWithName struct {
+	Name    string `protobuf:"bytes,1,opt,name=name,proto3" json:"name,omitempty"`
+	Message proto.Message
+}
+
+func (p jsonMarshalProtoWithName) MarshalJSON() ([]byte, error) {
+	strSer, serr := ConvertGolangProtoToJSONByGolangJSONPB(p.Message)
+	if serr != nil {
+		adscLog.Warnf("Error for marshaling [%s]: %v", p.Name, serr)
+		return []byte(""), serr
+	}
+	serialItem := []byte("{\"" + p.Name + "\":" + strSer + "}")
+	return serialItem, nil
+}
+
 var adscLog = log.RegisterScope("adsc", "adsc debugging", 0)
 
+// ConvertGolangProtoToJSONByGolangJSONPB help to implement the conversion from Proto message to string
+// Note: this conversion is just for golang protobuf by golang jsonpb
+func ConvertGolangProtoToJSONByGolangJSONPB(obj proto.Message) (string, error) {
```

Comment:
we can use protomarshal.ToJSONWithIndent

Reply:
okay, I will refactor it later